lib/random.php

```diff
@@ -296,14 +273,15 @@ function random_int($min, $max)
              * Apply mask
              */
             $val &= $mask;
+            $val += $valueShift;
```

Comment:
Yep, that's the reasoning. I think it ended up here from the logic we saw in logic written by @CodesInChaos https://gist.github.com/CodesInChaos/03f9ea0b58e8b2b8d435